core/modules/css/CssAccess.h

```diff
@@ -0,0 +1,422 @@
+/*
+ * LSST Data Management System
+ * Copyright 2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_CSS_CSSACCESS_H
+#define LSST_QSERV_CSS_CSSACCESS_H
+
+// System headers
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+// Third-party headers
+
+// LSST headers
+
+// Qserv headers
+#include "css/MatchTableParams.h"
+#include "css/NodeParams.h"
+#include "css/PartTableParams.h"
+#include "css/StripingParams.h"
+#include "css/TableParams.h"
+
+namespace lsst {
+namespace qserv {
+namespace css {
+
+class EmptyChunks;
+class KvInterface;
+
+/// @addtogroup css
+
+/**
+ *  @ingroup css
+ *
+ *  @brief C++ source file code template.
+ *
+ *  Class which provides access to information in Qserv Central State System.
+ *
+ *  This class is a wrapper around key-value storage (KvInterface) instance
+ *  with knowledge about structure of CSS keys, packing/unpacking mechanism,
+ *  etc.
+ *
+ *  This class  also provides information about empty chunk list. This will
+ *  likely migrate to some different interface (e.g. become a part of
+ *  secondary index) so it should be considered as temporary here.
+ *
+ *  This is a concrete class, instances can be copied around and all copies
+ *  share the same KvInterface instance (and empty chunk list).
+ */
+
+class CssAccess {
+public:
+
```

Comment:
I can why it can be attractive but it's not very common convention. I hope with `auto` the need for  these typedefs will be reduced.